### PR TITLE
Synchronize stcontrol.lua

### PR DESCRIPTION
**Pull request by dvdvideo1234**
Added: Zero crossing and margin support
Utilized: `Vector()` and `Unpack()`
Removed: The rest of the deprecated functions

### CODE REVIEW

**Comment by dvdvideo1234** [comment]
Same as: https://github.com/wiremod/wire-extras/pull/136

File: lua/entities/gmod_wire_expression2/core/custom/stcontrol.lua [hunk 1]
```diff
@@ -573,79 +600,79 @@ end
 
 __e2setcost(3)
 e2function array stcontrol:getGain()
-	if not this then return {0,0,0} end
+	if(not this) then return {0,0,0} end
```
Comment (Denneisk) [inline lua/entities/gmod_wire_expression2/core/custom/stcontrol.lua:603]: I don't see the purpose in adding these parentheses.

Comment (dvdvideo1234) [reply lua/entities/gmod_wire_expression2/core/custom/stcontrol.lua:603]: It is prettier this way. And 1:1 with the main repository.

Comment (thegrb93) [reply lua/entities/gmod_wire_expression2/core/custom/stcontrol.lua:603]: It's prettier for C users, but not lua, but this is your own library so I don't mind it

**Pull request merged**